metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/ExternalFunctions.java

```diff
@@ -0,0 +1,292 @@
+import java.io.PrintWriter;
+import java.util.List;
+import java.lang.ProcessBuilder;
+import java.lang.ClassLoader;
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.regex.Pattern;
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import org.apache.metron.common.dsl.Context;
+import org.apache.metron.common.dsl.StellarFunction;
+import org.apache.metron.common.dsl.ParseException;
+import org.apache.metron.common.dsl.Stellar;
+
+/**
+ * Executes external script on server via stellar process
+ */
+public class ExternalFunctions {
+
+	public static class ExecuteScript implements StellarFunction {
+
+        private ThreadedStreamHandler inStream;
+        private ThreadedStreamHandler errStream;
+        private boolean isOnTheList = false;
+
+        @Stellar(name="EXEC_SCRIPT",
+                description = "Executes an external shell function via stellar.",
+                params = {
+                        "exec - the executing cmd (ie. bash, sh, python)",
+                        "name - name of the script, located in /scripts " +
+                                "Do NOT include any special chars except(_), Do include file extension"
+                },
+                returns = "the return value of the function"
+        )
+
+	    @Override
+        public Object apply(List<Object> args, Context context) throws ParseException {
+            String exec = "";
+            String name = "";
+            String path = "";
+
+            // if args are provided, get args, only if in whitelist
+            if (args.size() >= 1) {
+                Object execObj = args.get(0);
+                if (!(execObj instanceof String)) { //check if string
+                    return null;
+                }
+                else if (((String) execObj).length() > 0) {
+                    exec = (String) execObj;
+                }
+                else {
+                    return null;
+                }
+
+                Object nameObj = args.get(1);
+                if (!(nameObj instanceof String)) { //check if string
+                    return null;
+                }
+                else if (((String) nameObj).length() > 0) {
+                    name = (String) nameObj;
+                }
+                else {
+                    return null;
+                }
+
+                if (!Pattern.matches("[0-9A-Za-z.]+", name)) {
+                    return null; //if not on whitelist
+                }
+
+                path = "/scripts" + name;
+                try {
+                    File script = new File(path);
+                    if (!script.exists() || script.isDirectory()) {
+                        return null;
+                    }
+                }
+                catch (NullPointerException e)  {
+                    System.err.println("Error: " + e.toString());
+                    return null;
+                }
```

Comment:
Rather than returning nulls, here and in many places throughout the code, it is better to actually throw meaningful exceptions, with explanatory text so that the user (assuming benign intent) has some ability to fix their mistakes.